### PR TITLE
Add custom header for staging copy and build script

### PR DESCRIPTION
**Pull request by krilnon**
As discussed in some of the docs workgroup meetings, it's useful for us to have a staging copy of the book.

Fixes https://github.com/apple/swift-book/issues/34

### CODE REVIEW

File: bin/update-book-preview [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# Check out the 'gh-pages' branch in a subdirectory.
+git worktree add --checkout gh-pages "$REMOTE"/gh-pages
+
+# Pretty print DocC JSON output, so that it has a stable ordering and
+# can be diffed.  This lets us determine whether rebuilding resulted in
+# any changes to the content.
+export DOCC_JSON_PRETTYPRINT="YES"
+
+# Build documentation, writing output in the gh-pages/docs subdirectory.
+export SWIFTPM_ENABLE_COMMAND_PLUGINS=1
```
Comment (ethan-kusters) [inline bin/update-book-preview:46]: Nit: This shouldn't be needed anymore.

```suggestion
```

File: bin/update-book-preview [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+swift package \
+    --allow-writing-to-directory "./gh-pages/docs" \
+    generate-documentation \
+    --target TSPL \
+    --disable-indexing \
+    --transform-for-static-hosting \
```
Comment (ethan-kusters) [inline bin/update-book-preview:52]: Nit: Static hosting transformation happens by default now.

```suggestion
```

**Review by ethan-kusters — APPROVED** [review]

**Review by mportiz08 — APPROVED** [review]
The custom header looks good from a functional perspective 👍

I have some minor suggestions on the styling but I think I should leave that for a separate PR :)

**Comment by amartini51** [comment]
> I have some minor suggestions on the styling but I think I should leave that for a separate PR :)

@mportiz08 If it's minor feedback, probably just as easy for us to discuss and incorporate style changes on this PR.

File: bin/update-book-preview (hunk 1 above)
Comment (amartini51) [reply bin/update-book-preview]: Removed it 0d56de46.

File: bin/update-book-preview (hunk 2 above)
Comment (amartini51) [reply bin/update-book-preview]: Removed it 0d56de46.

**Comment by amartini51** [comment]
Now that we're adding a custom footer too (#88) we also need to add some kind of switching mechanism so the header gets used only for previews.  Without the footer, we were ok only setting `--experimental-enable-custom-templates` for previews.

**Pull request merged**